nodebuilder/blob/blob.go

```diff
@@ -34,15 +34,52 @@ type Module interface {
 	// Included checks whether a blob's given commitment(Merkle subtree root) is included at
 	// given height and under the namespace.
 	Included(_ context.Context, height uint64, _ share.Namespace, _ *blob.Proof, _ blob.Commitment) (bool, error)
+	// GetCommitmentProof generates a commitment proof for a share commitment.
+	GetCommitmentProof(
+		ctx context.Context,
+		height uint64,
+		namespace share.Namespace,
+		shareCommitment []byte,
+	) (*blob.CommitmentProof, error)
 }
```

Comment:
The comment Rene referred to is about combining this with `GetProof`. As we decided to integrate APIs Blobstream needs into the node nicely, we should also the same for `GetProof`. I am open to doing that in a separate PR though if you think that's better @rach-id, but as long we release these two PRs together.

Reply:
The tricky part is unifying the new `CommitmentProof` structure with an existing `Proof` and making sure that `Included` method support it as well

Reply:
I prefer to do it in a separate PR. This PR has been open for so long, I don't think we should delay merging it any more. Hopefully, after handling the feedback above, we can go ahead and merge it. 

Created this issue for this: https://github.com/celestiaorg/celestia-node/issues/3582